crates/core/component/auction/src/component/dutch_auction.rs

```diff
@@ -202,155 +437,43 @@ trait Inner: StateWrite {
-
-    let prev_step_index = distance_from_start
-        .checked_div(step_size)
-        .ok_or_else(|| anyhow::anyhow!("step size is zero"))?;
-
-    if prev_step_index >= step_count {
-        return Ok(None);
-    }
-
-    let next_step_index = prev_step_index
-        .checked_add(1)
-        .ok_or_else(|| anyhow::anyhow!("step index has overflowed"))?;
-
-    let next_step_size_from_start = step_size.checked_mul(next_step_index).ok_or_else(|| {
-        anyhow::anyhow!(
-            "next step size from start has overflowed (step_size={}, next_step_index={})",
-            step_size,
-            next_step_index
-        )
-    })?;
-
-    Ok(start_height.checked_add(next_step_size_from_start))
-}
-struct TriggerData {
-    pub start_height: u64,
-    pub end_height: u64,
-    pub current_height: u64,
-    pub step_count: u64,
-}
-
-#[cfg(test)]
-mod tests {
-    use crate::component::dutch_auction::{compute_next_trigger, TriggerData};
-
-    #[test]
-    fn test_current_height_equals_start_height() {
-        let data = TriggerData {
-            start_height: 100,
-            end_height: 200,
-            step_count: 5,
-            current_height: 100,
-        };
-        assert_eq!(compute_next_trigger(data).unwrap(), Some(120));
-    }
-
-    #[test]
-    fn test_current_height_equals_end_height() {
-        let data = TriggerData {
-            start_height: 100,
-            end_height: 200,
-            step_count: 5,
-            current_height: 200,
-        };
-        assert_eq!(compute_next_trigger(data).unwrap(), None);
-    }
-
-    #[test]
-    fn test_current_height_below_start_height() {
-        let data = TriggerData {
-            start_height: 100,
-            end_height: 200,
-            step_count: 5,
-            current_height: 90,
-        };
-        assert_eq!(compute_next_trigger(data).unwrap(), Some(120));
-    }
-
-    #[test]
-    fn test_current_height_above_end_height() {
-        let data = TriggerData {
-            start_height: 100,
-            end_height: 200,
-            step_count: 5,
-            current_height: 210,
-        };
-        assert_eq!(compute_next_trigger(data).unwrap(), None);
-    }
-
-    #[test]
-    fn test_current_height_below_boundary() {
-        let data = TriggerData {
-            start_height: 100,
-            end_height: 200,
-            step_count: 5,
-            current_height: 119,
-        };
-        assert_eq!(compute_next_trigger(data).unwrap(), Some(120));
-    }
-
-    #[test]
-    fn test_current_height_above_boundary() {
-        let data = TriggerData {
-            start_height: 100,
-            end_height: 200,
-            step_count: 5,
-            current_height: 121,
-        };
-        assert_eq!(compute_next_trigger(data).unwrap(), Some(140));
-    }
+/// TODO(erwan): prove that this works. More work needed to work out validation
+/// rules that let us operate within the 112 bits constraint.
```

Comment:
it feels like this should be something we can check upfront when we parse the auction description for the first time?

Reply:
Yes for sure, basically the same validation rules for position apply for a DA description.

Reply:
if we require `input_amount`, `output_amount` < 52 bits and `step_count < 256` then we are guaranteed that `input_amount * output_amount * step_count` < 112 bits.

Reply:
I put a TODO to think the case of large denom units (e.g. $10^{18} ~ 60 bits$)

Reply:
we could exclude those from auctions, seems fine